### ISSUE TITLE
approxInnerStrongholdRing does not have claimed accuracy

### ISSUE THREAD

**Issue by orlp** [Cubitect/cubiomes#46]
In the documentation of of `approxInnerStrongholdRing` we find:

```
 * Note that this function requires only the lower 48-bits of the seed, and the
 * accuracy is within +/-112 blocks.
```

But I've found this not to be true. E.g.:

```c
#include <stdio.h>
#include "finders.h"

int main(int argc, char** argv) {
    int mcversion = MC_1_16;
    initBiomes();
    LayerStack g;
    setupGenerator(&g, mcversion);


    int64_t seed = 707347623312870631;
    Pos p[3];

    approxInnerStrongholdRing(p, mcversion, seed);
    for (int i = 0; i < 3; ++i) {
        p[i].x = (p[i].x << 4) + 8;
        p[i].z = (p[i].z << 4) + 8;
    }
    printf("Approx: (%d, %d) (%d, %d) (%d, %d)\n", p[0].x, p[0].z, p[1].x, p[1].z, p[2].x, p[2].z);

    applySeed(&g, seed);
    findStrongholds(mcversion, &g, NULL, p, seed, 3, 1);
    printf("Exact: (%d, %d) (%d, %d) (%d, %d)\n", p[0].x, p[0].z, p[1].x, p[1].z, p[2].x, p[2].z);

    return 0;
}
```

prints

```
Approx: (-216, -1528) (2424, 968) (-1144, 920)
Exact: (-116, -1532) (1736, 540) (-1160, 936)
```

Note that the second stronghold is 800 blocks away from its approximated location.

**Comment by Badel2** [comment]
I think that the issue is that stronghold generation uses the function `findBiomePosition`, which modifies the RNG seed. Therefore, only the approx coordinates of the first stronghold will be correct. The second and the third strongholds will be at the correct angle, but wrong distance from (0, 0). So if you go in a straight line from (0, 0) to the approx coordinates, the stronghold will be within +/- 112 blocks from any point of that line.

In your case it's a coincidence that the coordinates of the third stronghold are pretty close, but if you try other seeds you will find that only the first stronghold matches.

There are some exceptions, when `findBiomePosition` does not find any valid biome it does not modify the seed and the approx coordinates of the next stronghold are correct. In this context, valid biomes are biomes with "height > 0.0". For example, if the first stronghold is in the middle of the ocean, the coordinates of the second stronghold will be correct.

If I did not miss anything, this has always been like that, even in old versions of minecraft before stronghold rings.

TL;DR: we need to update the documentation of `approxInnerStrongholdRing` and say that only the first stronghold will be accurate.

**Comment by Cubitect** [comment]
That was my bad. For some reason I totally overlooked that the distance also depends on RNG from the biome check, and for my test seeds it happened to work out, so I didn't catch this. 

The point of having this inner ring approximation is kind of mute because of this. I replaced the function with a kind of stronghold iterator. That should hopefully provide the flexibility I was going for and minimize processing to what's required:

```C
#include "finders.h"
#include <stdio.h>

int main() {
    initBiomes();
    
    int64_t s = 707347623312870631;
    int i, mc = MC_1_8;
    
    StrongholdIter sh;
    // get approximation of first stronghold only
    Pos a = initFirstStronghold(NULL, mc, s);
    printf("first approx: (%6d %6d)\n", a.x, a.z);
    
    LayerStack g;
    setupGenerator(&g, mc);
    applySeed(&g, s);
    
    // iterate over all strongholds, generating the locations as we go
    initFirstStronghold(&sh, mc, s);
    while (nextStronghold(&sh, &g, NULL)) {
        printf("accurate:  (%6d %6d)", sh.pos.x, sh.pos.z);
        printf(" next approx: (%6d %6d)\n", sh.nextapprox.x, sh.nextapprox.z);
    }
    
    Pos p[3];
    findStrongholds(mc, &g, NULL, p, s, 3, 1);
    for (i = 0; i < 3; i++) {
        printf("reference: (%6d %6d)\n", p[i].x, p[i].z);
    }

    return 0;
}
```
prints
```
first approx: (   -88   -680)
accurate:  (   -80   -568) next approx: (   744    296)
accurate:  (   716    240) next approx: (  -808    664)
accurate:  (  -920    552) next approx: (    88   -888)
reference: (   -80   -568)
reference: (   716    240)
reference: (  -920    552)
```
